aiida_quantumespresso/common/types.py

```diff
@@ -31,3 +31,12 @@ class SpinType(enum.Enum):
     COLLINEAR = 'collinear'
     NON_COLLINEAR = 'non_collinear'
     SPIN_ORBIT = 'spin_orbit'
+
+
+class RestartType(enum.Enum):
+    """Enumeration of ways to restart a calculation in Quantum ESPRESSO."""
+
+    FULL = 'full'
+    FROM_SCRATCH = 'from_scratch'
+    FROM_CHARGE_DENSITY = 'from_charge_density'
+    FROM_WAVE_FUNCTIONS = 'from_wave_functions'
```

Comment:
Hi guys, I wanted just to point out that it could be useful to have also `FROM_FILES` (or something like that), so that one can restart from the density and wave functions. I know that 'full' is meant to do so, but it will also read the atomic positions.

Reply:
My first inclination would be to say that restarting from the wave functions, it should be pretty fast for QE to recalculate the charge density. But if I remember correctly, QE doesn't actually do this, instead just plugging in the wave functions but for the potential calculated from the atomic charge density, correct?

Reply:
Yes, should be like this. Two case scenario would be:
- electric enthalpy routine: it uses the wfcs to build the polarization operator (essentially you start with a slightly different hamiltonian)
- maybe in some restart when vc-relaxing?

But I do agree at the end it is not such a difference. I was just wondering, since the new implementation is so cool (great job!!! :D ), it would be quite easy to just add that one more, just in case one needs it. 
Or do you think that the "experienced" user can still always tweak the inputs if that is the case?

Reply:
Thanks for the clarification, @bastonero! Note that the restart types are only used for setting the restarts after an error has been handled using the `set_restart_type` method. So if a user wants to restart from a previous calculation, the correct inputs have to be provided, not the restart type. 

Happy to implement another restart type (not sure about the name, `FROM_FILES` seems a bit too general, maybe `FROM_CHARGE_AND_WFC`?). Do you think there is already a current error handler where this restart type would be used though?

Reply:
Ah ok, I see, thanks! For the moment, actually, I still don't have an example of handler where it would be useful. `FROM_CHARGE_AND_WFC` sounds good anyway.